File: C/42. Trapping Rain Water.c
int trap(int* height, int heightSize){
    int sol=0;
    int l[heightSize];
    int r[heightSize];
    l[0]=height[0];
    r[heightSize-1]=height[heightSize-1];
    for(int i=1; i<heightSize; i++){
        l[i]= fmax(l[i-1],height[i]);
    }
    for(int i= (heightSize-2); i>=0; i--){
        r[i]= fmax(r[i+1],height[i]);
    }
    for(int i=0; i<heightSize; i++){
        sol+= (fmin(l[i], r[i]))-height[i];
    }
    return sol;
}

File: C#/42. Trapping Rain Water.cs
public class Solution {
    public int Trap(int[] height) {
        int n= height.Length;
        int sol=0;
        int[] l=new int[n];
        int[] r=new int[n];
        l[0]=height[0];
        r[n-1]=height[n-1];
        for(int i=1; i<n; i++){
            l[i]= Math.Max(l[i-1],height[i]);
        }
        for(int i=n-2; i>=0; i--){
            r[i]= Math.Max(r[i+1],height[i]);
        }
        for(int i=0; i<n; i++){
            sol+= (Math.Min(l[i], r[i]))-height[i];
        }
        return sol;
    }
}
